src/Planner/PlannerJoinTree.cpp

```diff
@@ -606,6 +607,93 @@ std::unique_ptr<ExpressionStep> createComputeAliasColumnsStep(
     return alias_column_step;
 }
 
+std::unique_ptr<ExpressionStep> createAddingVirtualColumnsStep(
+    const DataStream & current_data_stream,
+    const StoragePtr & storage)
+{
+    ColumnWithTypeAndName column;
+    column.name = "_table";
+    column.type = std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>());
+    column.column = column.type->createColumnConst(0, storage->getStorageID().getTableName());
+
+    auto adding_virtual_column_dag = ActionsDAG::makeAddingColumnActions(std::move(column));
+    auto adding_virtual_column_step = std::make_unique<ExpressionStep>(current_data_stream, std::move(adding_virtual_column_dag));
+    adding_virtual_column_step->setStepDescription("Add universal virtual columns");
+    return adding_virtual_column_step;
+}
+
+bool extractRequiredColumnsFromStorage(
```

Comment:
Here as well

Reply:
Rename "extractRequiredColumnsFromStorage" to "extractRequiredNonTableColumnsFromStorage"